### DISCUSSION TITLE
Left Alt, single-key special characters like in Mac OS in Windows

### DISCUSSION THREAD

**Discussion by Ariane-B** [Dygmalab/Bazecor#438] [Q&A]
I normally use the Canadian Multilingual Standard. In this layout, if, say, I want to type a curly bracket `{`, I'll need to do `LCtrl`+`LAlt`+`7` or `AltGr`+`7`. It's the same for a bunch of other special characters handy for programmers and French speakers (and I happen to be both), such as ~`|«»°[]<>.

That's fine, I guess, but when you're a programmer and constantly typing such characters, it can get a little annoying.

The same keyboard layout on Mac OS has exactly the same shortcuts, except you can trigger them with just `LAlt`+`key`. That's a pretty big effort (and wrist pain) saver. Up until now, I've been using Microsoft PowerToys's Keyboard Editor to set up "shortcuts" that translate `Alt`+`key` into `Ctrl`+`Alt`+`key`. It works fine most of the time, but PowerToys isn't 100% stable, and also, it's a pain to have to set it up whenever I switch PCs or I need to reformat my PC because... Microsoft reasons I guess.

So, I thought maybe I could set that up directly into Bazecor... But I can't seem to find a way to do that. Am I missing something, or is that just not possible? And if not, is it planned in the future?

**Comment by GazHank** [comment]
Not being familiar with the Canadian layout (or how it works between the different operating systems), can I check if I've understood how it is working and what re-mapping you are using in PowerToys Keyboard Manager?

If I understand correctly, the default behaviour is that to type a left curly bracket `{` you need to press:
Mac: `Left Alt + 7`
Win: `Right Alt + 7` or `Ctrl + Left Alt + 7`

It kind of sounds like you would like the Left Alt key to operate like the Right Alt (AltGR) key instead?  or to swap those alt keys?  but I assume that's not how you are remapping them in PowerToys

Are you trying to have the Left Alt key behave differently depending on the key it is used in combination with? something like:
`Left Alt + 6` = `Left Alt + 6`
`Left Alt + 7` = `Right Alt + 7`

If so then I don't think this type of remapping is easy to achieve, but you could setup the Left Alt as a Superkey, where Holding the key behaves like Left Alt, and tap then hold behaves like Right Alt.  Would then work?

**Comment by GazHank** [comment]
If you haven't seen it already you might want to check out one of the Dygma Labs videos from a few weeks ago https://youtu.be/HnBLoZGkPgc and there also seem to be some useful comments on the video too (I didn't even know that useful comments were possible on youtube videos!)

**Comment by Ariane-B** [comment]
I tried telling the LAlt key to behave like RAlt when held. The problem is that then it breaks keyboard shortcuts that expect LAlt, such as Alt+Tab. :( So I reverted to the PowerToys method. 

Right now I'm out and about and checking out a video would be difficult but I'll be sure to check it out later.

**Comment by GazHank** [comment]
I've tried using the superkey functionality myself and it just doesnt seem to function correctly, so I suspect the SuperKey functionality is just a bit too buggy at the moment.

In terms of future roadmap options, it looks like the [chord plugin](https://github.com/keyboardio/Kaleidoscope/tree/master/plugins/Kaleidoscope-Chord) might in the future be a way to enable this functionality, but that would need both firmware and bazecor changes to make it work.

In the meantime, another option you might want to try would be to remap the right alt key to somewhere more convenient; but that would still mean that you are using different keyboard combos for mac and windows.  Or perhaps the more extreme option would be use enable two keyboard layouts in the Operating Systems, your usual canadian for general use and perhaps a UK or US ISO version for coding (you could use the website http://kbdlayout.info/KBDCAN/ compare similar language layouts to see if there is one that brings all of your most used keys to the surface and avoids painful key combos)

**Comment by Ariane-B** [comment]
That does sound promising. As for changing keyboard layouts... Well, sounds led convenient than the PowerToys solution, sadly. Really appreciate the advice though!
